src/Quartz/Job/DirectoryScanJob.cs

```diff
@@ -57,95 +66,52 @@ public DirectoryScanJob()
         /// the job will use during execution.</param>
         public void Execute(IJobExecutionContext context)
         {
-            JobDataMap mergedJobDataMap = context.MergedJobDataMap;
-            SchedulerContext schedCtxt;
-            try
-            {
-                schedCtxt = context.Scheduler.Context;
-            }
-            catch (SchedulerException e)
-            {
-                throw new JobExecutionException("Error obtaining scheduler context.", e, false);
-            }
-
-            string dirName = mergedJobDataMap.GetString(DirectoryName);
-            string listenerName = mergedJobDataMap.GetString(DirectoryScanListenerName);
-
-            if (dirName == null)
-            {
-                throw new JobExecutionException("Required parameter '" +
-                                                DirectoryName + "' not found in merged JobDataMap");
-            }
-            if (listenerName == null)
-            {
-                throw new JobExecutionException("Required parameter '" +
-                                                DirectoryScanListenerName + "' not found in merged JobDataMap");
-            }
-
-            object temp;
-            schedCtxt.TryGetValue(listenerName, out temp);
-            IDirectoryScanListener listener = (IDirectoryScanListener) temp;
-
-            if (listener == null)
-            {
-                throw new JobExecutionException("DirectoryScanListener named '" +
-                                                listenerName + "' not found in SchedulerContext");
-            }
+            DirectoryScanJobModel model = DirectoryScanJobModel.GetInstance(context);
 
-            DateTime lastDate = DateTime.MinValue;
-            if (mergedJobDataMap.ContainsKey(LastModifiedTime))
+            List<FileInfo> updatedFiles = new List<FileInfo>();
```

Comment:
List is not a thread-safe collection, should either use a concurrent collection or explicit lock object.